### PR TITLE
[fix] Retry flaky test

### PR DESCRIPTION
**Pull request by kciesielski**
This test occasionally fails, because it connects to httpbin.org

Before submitting pull request:
- [x] Check if the project compiles by running `sbt compile`
- [x] Verify docs compilation by running `sbt compileDocs`
- [x] Check if tests pass by running `sbt test`
- [x] Format code by running `sbt scalafmt`

### CODE REVIEW

File: core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala [hunk 1]
```diff
@@ -189,6 +191,17 @@ abstract class StreamingTest[F[_], S]
     val numChunks = 100
     val url = uri"https://httpbin.org/stream/$numChunks"
 
+    implicit val monadError: MonadError[Future] = new FutureMonad
+
+    def retryImmediatelyOnError[A](action: => Future[A], retriesLeft: Int): Future[A] =
+      action.handleError { case error =>
```
Comment (katlasik) [inline core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala:197]: **Nitpick**: do you need case here?

**Review by katlasik — APPROVED** [review]
lgtm

File: core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala (hunk 1 above)
Comment (kciesielski) [reply core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala:197]: Yes, at first I wrote without case, but needed to add it for compatibility with Scala.js https://github.com/softwaremill/sttp/pull/1874/commits/57724383e90475e41bc4748cfc7c0997efb9e67b

File: core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala [hunk 2]
```diff
@@ -208,6 +221,7 @@ abstract class StreamingTest[F[_], S]
 
         urlRegex.findAllIn(responseBody).length shouldBe numChunks
       }
+    retryImmediatelyOnError(runTest, retriesLeft = 5)
```
Comment (adamw) [inline core/src/test/scala/sttp/client4/testing/streaming/StreamingTest.scala:224]: sorry for commenting on merged PRs, but I'd add a comment here why this particular tests is "fixed" by adding a repeat :)

**Pull request merged**